### PR TITLE
Fix Possible DISPATCH_SEND Inline Issue

### PR DESCRIPTION
**Pull request by nibanks**
## Description

We've seen indication of a possible issue when sending at DISPATCH_LEVEL, so we're fixing the behavior to always queue in these cases (no inline execution).

## Testing

Updated some tests to send at DISPATCH_LEVEL in kernel.

## Documentation

N/A

### CODE REVIEW

File: src/core/api.c [hunk 1]
```diff
@@ -1083,8 +1083,11 @@ MsQuicStreamSend(
     SendRequest->ClientContext = ClientSendContext;
 
     SendInline =
-        !Connection->Settings.SendBufferingEnabled &&
-        Connection->WorkerThreadID == CxPlatCurThreadID();
+        !Connection->Settings.SendBufferingEnabled
+#ifdef CXPLAT_AT_DISPATCH
```
Comment (csujedihy) [inline src/core/api.c:1087]: why not have a dummy macro for other platforms?

Comment (nibanks) [reply src/core/api.c:1087]: I can, but this required fewer files to be changed. That's why I did this instead.

Comment (csujedihy) [reply src/core/api.c:1087]: I think only 2 headers needed to be modified. We should be consistent with what we did for CXPLAT_PASSIVE_CODE and CXPLAT_IRQL.

**Comment by csujedihy** [comment]
We have the same inline execution for other APIs like StreamClose and ConnectionClose. Should we change them as well?

File: src/core/api.c (hunk 1 above)
Comment (nibanks) [reply src/core/api.c]: Fixed

**Comment by nibanks** [comment]
> We have the same inline execution for other APIs like StreamClose and ConnectionClose. Should we change them as well?

No, because *Close functions are required to be called at PASSIVE_LEVEL.

**Review by csujedihy — APPROVED** [review]

**Review by csujedihy — APPROVED** [review]

**Pull request merged**